apps/builder/src/features/editor/components/TypebotHeader/TypebotHeader.tsx

```diff
@@ -52,6 +52,39 @@ export const TypebotHeader = () => {
     setRightPanel(RightPanel.PREVIEW)
   }
 
+  const isUndoShortcut = (event: KeyboardEvent) => {
+    const key = event.key.toLowerCase()
+    const cmdHeld = event.metaKey
+    const ctrlHeld = event.ctrlKey
+    const Z = 'z'
+
+    return (cmdHeld && key === Z) || (ctrlHeld && key === Z)
+  }
+
+  useEffect(() => {
+    document.addEventListener(
+      'keydown',
+      (event: KeyboardEvent) => {
+        if (isUndoShortcut(event) && canUndo) {
+          undo()
+        }
+      },
+      false
+    )
+
+    return () => {
+      document.removeEventListener(
+        'keydown',
+        (event: KeyboardEvent) => {
+          if (isUndoShortcut(event) && canUndo) {
+            undo()
+          }
+        },
+        false
+      )
+    }
+  }, [canUndo, undo])
+
```

Comment:
Could you refactor this to a `useUndoShortcut` hook that we would use like this:

```ts
useUndoShortcut(undo)
```

Reply:
added useUndoShortcut hook